### PR TITLE
Added type annotations

### PR DESCRIPTION
**Pull request by jidroid404**
Referencing the issue #[1456](https://github.com/OpenMined/PySyft/issues/1456)

### CODE REVIEW

File: syft/grid.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
 from collections import Counter
-
+from typing import Tuple,Dict
```
Comment (robert-wagner) [inline syft/grid.py]: Import these on separate lines

**Comment by robert-wagner** [comment]
Also please run `black syft`

**Comment by jidroid404** [comment]
Not familiar with black, did the pip install and `black syft`
Help me out please
Screenshot from 2019-06-03 07-58-13

**Comment by jidroid404** [comment]
@mari-linhares I don't know how to add more files to this commit (embarrassing Ik), I already end up making a new PR instead of adding commits to this one. What's the git command?

**Comment by mari-linhares** [comment]
Hey @jidroid404, 

You don't need to add more files to the commit, you can modify the file locally, create a new commit and then push to your local branch (this PR will be automatically updated), here are the commands:

```
# go to your local copy of this repo
cd <path where PySyft was cloned> 
# update your local branch
git pull origin dev
# modify syft/grid.py as suggested 
...
# add changes
git add syft/grid.py
# commit changes
git commit
# push changes
git push origin dev
```

**Comment by jidroid404** [comment]
cmomit
Resolved conflict !

**Comment by robert-wagner** [comment]
something in your directory is not added. If you run git status it will tell you what it is

**Comment by mari-linhares** [comment]
For ignoring some file you can add all the files you want to commit then run `git stash`. Careful: this will undo any changes to any file not staged (added).

**Comment by jidroid404** [comment]
Screenshot from 2019-06-03 18-59-32
Now this !

**Comment by mari-linhares** [comment]
try: `git pull origin dev`

File: examples/tutorials/Part 1 - The Basic Tools of Private Deep Learning.ipynb [hunk 2]
```diff
@@ -83,7 +83,11 @@
    "source": [
     "# Run this cell to see if things work\n",
     "import syft as sy\n",
+<<<<<<< HEAD
```
Comment (robert-wagner) [inline examples/tutorials/Part 1 - The Basic Tools of Private Deep Learning.ipynb]: Please remove these lines. The indicate where there was a merge conflict

Comment (robert-wagner) [reply examples/tutorials/Part 1 - The Basic Tools of Private Deep Learning.ipynb]: On second thought this file should not be in this pr.
Please run `git checkout dev -- 'examples/tutorials/Part 1 - The Basic Tools of Private Deep Learning.ipynb'`
Then commit the changes

**Comment by jidroid404** [comment]
Getting `nothing to commit` on doing so.

**Comment by robert-wagner** [comment]
> Getting `nothing to commit` on doing so.

I forgot this is in a fork. Let me see if there is an easy way to do this

**Comment by jidroid404** [comment]
> Deep Learning.ipynb'

`fatal: invalid reference: upstream/dev` after running `git checkout`

**Comment by robert-wagner** [comment]
@jjdroid404 did adding the remote work?

**Comment by jidroid404** [comment]
@robert-wagner Ya that worked.

File: syft/frameworks/torch/functions.py [hunk 3]
```diff
@@ -1,7 +1,8 @@
 from .tensors.interpreters import MultiPointerTensor
+from syft.workers import VirtualWorker
 
 
-def combine_pointers(*pointers):
+def combine_pointers(*pointers: VirtualWorker) -> MultiPointerTensor:
```
Comment (robert-wagner) [inline syft/frameworks/torch/functions.py]: pointers should be a list of pointers not a virtual worker

Comment (jidroid404) [reply syft/frameworks/torch/functions.py]: What is it to be given as type then?

File: syft/frameworks/torch/functions.py [hunk 4]
```diff
@@ -1,7 +1,9 @@
 from .tensors.interpreters import MultiPointerTensor
+from syft.frameworks.torch.pointers.object_pointer import ObjectPointer
+from typing import List
 
 
-def combine_pointers(*pointers):
+def combine_pointers(*pointers: List[ObjectPointer]) -> MultiPointerTensor:
```
Comment (robert-wagner) [inline syft/frameworks/torch/functions.py]: you're missing a close paren

Comment (jidroid404) [reply syft/frameworks/torch/functions.py]: I don't see it, where?

Comment (jidroid404) [reply syft/frameworks/torch/functions.py]: Why isn't this being merged? What am I missing?

Comment (jvmancuso) [reply syft/frameworks/torch/functions.py]: The close paren is there, there's just a return type annotation that hides it a bit. Suggest resolving this thread

Comment (jidroid404) [reply syft/frameworks/torch/functions.py]: Well,that is how the formatter (black) formatted it. Should I change it then ? @jvmancuso

Comment (jvmancuso) [reply syft/frameworks/torch/functions.py]: No, I'm saying there's no change needed here, so @robert-wagner can clear this requested change when he's able

Comment (mari-linhares) [reply syft/frameworks/torch/functions.py]: @jidroid404 apparently this file still needs to be merged.

Comment (robert-wagner) [reply syft/frameworks/torch/functions.py]: My apologies @jjdroid404 . I read the line in correctly

**Pull request merged**